ncbi/src/main/java/org/reactome/release/EuropePMC.java

```diff
@@ -0,0 +1,375 @@
+				"</description>",
+				2
+			),
+			indentString("<email>help@reactome.org</email>", 2),
+			indentString("</provider>", 1),
+			"</providers>"
+		);
+	}
+
+	/**
+	 * Writes Europe PMC link file to pre-set output directory.  Pathway and literature reference data retrieved from
+	 * the graph database.
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @throws IOException Thrown if creating or appending for either file fails
+	 */
+	private void writeEuropePMCLinkFile(Session graphDBSession) throws IOException {
+		Path europePMCLinkFilePath = getEuropePMCLinkFilePath();
+
+		Files.deleteIfExists(europePMCLinkFilePath);
+		Files.createFile(europePMCLinkFilePath);
+
+		logger.info("Writing Europe PMC Link file");
+
+		appendWithNewLine(getXMLDeclaration(), europePMCLinkFilePath);
+		appendWithNewLine(getOpenRootTag(), europePMCLinkFilePath);
+
+		for (EuropePMCLink europePMCLink : fetchEuropePMCLinks(graphDBSession)) {
+			appendWithNewLine(europePMCLink.getLinkXML(), europePMCLinkFilePath);
+		}
+		appendWithNewLine(getCloseRootTag(), europePMCLinkFilePath);
+
+		logger.info("Finished writing Europe PMC Link file");
+	}
+
+	/**
+	 * Standard XML declaration header
+	 * @return XML declaration as String
+	 */
+	private String getXMLDeclaration() {
+		return "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>";
+	}
+
+	/**
+	 * Opening XML tag of Europe PMC link file
+	 * @return Opening XML tag as String
+	 */
+	private String getOpenRootTag() {
+		return "<" + rootTag + ">";
+	}
+
+	/**
+	 * Closing XML tag of Europe PMC link file
+	 * @return Closing XML tag as String
+	 */
+	private String getCloseRootTag() {
+		return "</" + rootTag + ">";
+	}
+
+	/**
+	 * Indents String by given number of tabs
+	 * @param string String to indent
+	 * @param numberOfIndents Number of times to indent string
+	 * @return Tab-indented string
+	 */
+	private static String indentString(String string, int numberOfIndents) {
+		return StringUtils.repeat("\t", numberOfIndents).concat(string);
+	}
+
+	/**
+	 * Returns Path for Europe PMC Profile File based on pre-set output directory and Reactome version
+	 * @return Path object for Europe PMC Profile File
+	 */
+	private Path getEuropePMCProfileFilePath() {
+		return Paths.get(outputDir, "europe_pmc_profile_reactome_" + reactomeVersion + ".xml");
+	}
+
+	/**
+	 * Returns Path for Europe PMC Link File based on pre-set output directory and Reactome version
+	 * @return Path object for Europe PMC Link File
+	 */
+	private Path getEuropePMCLinkFilePath() {
+		return Paths.get(outputDir, "europe_pmc_links_reactome_" + reactomeVersion + ".xml");
+	}
+
+	/**
+	 * Class to hold the relationship between a Reactome pathway and a literature reference.  The class can generate
+	 * a Europe PMC Link XML Node based on this relationship to be used in the exported Europe PMC Link File.
+	 * @author jweiser
+	 */
+	public static class EuropePMCLink {
```

Comment:
Is there a reason this is public? I thought inner classes usually had a more limited scope.

Reply:
I've kept it as an inner class because it does really belong as part of the EuropePMC class, but I've made it package-private since there is some functionality I need to test for the inner class itself.